tests/karate/src/test/java/common/net/MultiMsgWebSocketClient.java

```diff
@@ -41,113 +42,114 @@ public MultiMsgWebSocketClient(WebSocketOptions options, Logger logger, MessageQ
     setTextHandler(m -> true);
   }
 
-  public void send(Map<String, Object> jsonDataMap){
-    this.send(Json.of(jsonDataMap).toString());
-  }
-
-  @Override
-  public void signal(Object result) {
-    logger.trace("signal called: {}", result);
-    queue.onNewMsg(result.toString());
-  }
-
-  @Override
-  public synchronized Object listen(long timeout) {
-    logger.trace("entered listen wait state");
-    String msg = queue.take();
-
-    if (msg == null) logger.error("listen timed out");
-
-    return msg;
-  }
-
-  public MessageBuffer getBuffer() {
-    return queue;
+  /**
+   * JSON messages defined inside features are interpreted as maps from String to Object.
+   * This method is called directly inside features and is just a wrapper around the send method of WebsocketClient (that takes Strings).
+   * @param jsonRpcMsg the message to send as a JSON map
+   *                    (for example: subscribe and catchup in simpleScenarios.feature).
+   */
+  public void send(Map<String, Object> jsonRpcMsg){
+    this.send(mapToJsonString(jsonRpcMsg));
   }
 
-
-  public void publish(Map<String, Object> jsonDataMap, String channel){
-    Json dataJson = Json.of(jsonDataMap);
-    String data = dataJson.toString();
-    Random random = new Random();
-    int id = random.nextInt();
-    idAssociatedWithSentMessages.put(data, id);
-    Json request =  jsonConverter.publishMessageFromData(data, id, channel);
-    System.out.println("The final sent request is : " + request.toString());
-    this.send(request.toString());
+  /**
+   * JSON messages defined inside features are interpreted as maps from String to Object.
+   * This method is called directly inside features to send a complete publish message given the high-level message data to publish.
+   * @param highLevelMessageDataMap the high-level message to publish as a JSON map
+   *                                (for example: validCreateRollCall and badCreateRollCall in createRollCall.feature).
+   * @param channel the channel to publish on.
+   */
+  public void publish(Map<String, Object> highLevelMessageDataMap, String channel){
+    String highLevelMessageData = mapToJsonString(highLevelMessageDataMap);
+    int messageId = new Random().nextInt(Integer.MAX_VALUE);
```

Comment:
`Integer.MAX_VALUE` is already the default max value for nextInt so it's redundant